core/arch/x86/opcode.h

```diff
@@ -1438,4 +1438,11 @@ enum { /* FIXME: vs RAW_OPCODE_* enum */
 #define DEBUG_REGISTERS_FLAG_ENABLE_DR3 0xc0
 extern app_pc debugRegister[DEBUG_REGISTERS_NB];
 
+/* Tells if instruction will trigger an exception because of debug register. */
+static inline bool
+debug_register_fire_on_addr(app_pc pc) {
+    return (pc != NULL && (pc == debugRegister[0] || pc == debugRegister[1] ||
+                           pc == debugRegister[2] || pc == debugRegister[3]));
```

Comment:
So this assumes DEBUG_REGISTERS_NB is 4?  Seems best to have an assert for that.